src/Utils/NumUtils.cpp

```diff
@@ -102,4 +103,85 @@ uint64_t NumUtils::getMask(uint64_t wide) {
+ * a std::from_chars() for floating point numbers; available only >= g++-11
+ * for instance.
+ *
+ * So here we use the c++ detection pattern to determine if that function
+ * is available, otherwise fall back to slower best-effort implementation that
+ * copies it to a nul-terminated buffer, then calls the old strtof(), strotd()
+ * functions.
+ */
+template <typename T, typename = void>
+struct from_chars_available : std::false_type {};
+template <typename T>
+struct from_chars_available<
+    T, std::void_t<decltype(std::from_chars(
+           std::declval<const char *>(), std::declval<const char *>(),
+           std::declval<T &>()))>> : std::true_type {};
+
+template <typename T>
+inline constexpr bool from_chars_available_v = from_chars_available<T>::value;
+
+// Copy everything that looks like a number into output iterator.
+static void CopyNumberTo(const char *in_begin, const char *in_end,
+                         char *out_begin, const char *out_end) {
+  const char *src = in_begin;
+  char *dst = out_begin;
+  const char *extra_allowed = "+-.e";
+  bool have_point = false;
+  out_end -= 1;  // Allow space for 0 termination.
+  while (src < in_end && (isdigit(*src) || strchr(extra_allowed, *src)) &&
+         dst < out_end) {
+    // The sign is only allowed in the first character of the buffer
+    if ((*src == '+' || *src == '-') && dst != out_begin) break;
+    // only allow one decimal point
+    if (*src == '.') {
+      if (have_point)
+        break;
+      else
+        have_point = true;
+    }
+    *dst++ = *src++;
+  }
+  *dst = '\0';
+}
+
+template <typename T, T (*strto_fallback_fun)(const char *, char **)>
+static const char *convert_strto_ieee(std::string_view s, T *result) {
+  if constexpr (from_chars_available_v<T>) {
+    return internal::convert_strto_num<T>(s, result);
+  }
+
+  // Fallback in case std::from_chars() does not exist for this type. Here,
+  // we just call the corresponding C-function, but first have to copy
+  // the number to a local buffer, as that one requires \0-termination.
+  char buffer[64];
+
+  // Need to skip whitespace first to not use up our buffer for that.
+  std::string_view n = s;
+  while (!n.empty() && isspace(n.front())) n.remove_prefix(1);
+
+  CopyNumberTo(n.data(), n.data() + n.size(), buffer, buffer + sizeof(buffer));
+  char *endptr = nullptr;
+  *result = strto_fallback_fun(buffer, &endptr);
+  if (endptr == buffer) return nullptr;  // Error.
+
+  // Now, convert our offset back relative to the original string.
+  return n.data() + (endptr - buffer);
+}
+
+const char *convert_strtof(std::string_view s, float *result) {
+  return convert_strto_ieee<float, strtof>(s, result);
+}
+
+const char *convert_strtod(std::string_view s, double *result) {
+  return convert_strto_ieee<double, strtod>(s, result);
+}
+
+const char *convert_strtold(std::string_view s, long double *result) {
+  return convert_strto_ieee<long double, strtold>(s, result);
+}
```

Comment:
Just removing `convert_` would not work as `strotod()` etc are already standard library names which would be at best confusing.

Since we also provide the functionality of returning the position after the parsed value, how about something like `parse_float()`, `parse_int()` ? 

or `ParseAtoi()`, `ParseAtof()`  (leaning on atoi(), atof()...)

Reply:
Uh, what is Pascal case (haven't written Pascal in 30 years :) ). Is that CamelCase ? Yeah, casing was not really part of the suggestion, just the naming.

strTo32() is close to strtol() just with a weird casing in-between. I'd like to have something that is not accidentally mix-up-able with the standard library functions.

How about `parse_int32()`, `parse_uint32()`, ... in particular as our return value is particularly well-suited for parsing.

(also careful with strToF128() - `long double` is not necessarily a 128 bit type, typically `long double` is either just `double` or the weird 80 bit double in Intel machines; so all we can say would be long double)

Reply:
alright, renamed to to `parse_xxx()` for now and also provided overrides that allow to parse negative values into unsigned ints.

Reply:
I think the best would be that we always parse values with a `-` sign as signed and everything else as unsigned. And then we type-cast it to the final value.

That way, we get the best of both worlds: we parse the full range, but also we get the expected wrapping behavior.
Will do that on Monday.